model/operation.cc

```diff
@@ -435,16 +435,42 @@ namespace minsky
     switch (classify(type()))
       {
       case function: case reduction: case scan: case tensor:
-        if (check && !ports[1]->units(check).empty())
-          throw_error("function input not dimensionless");
-        return {};
+        switch (type())
+          {
+		  case percent:
+		    { 
+		     // Add % sign to units from input to % operator		
+             if (!ports[0]->wires().empty() && !ports[1]->wires().empty())  
+               {
+				 if (auto vV=dynamic_cast<VariableValue*>(&ports[0]->wires()[0]->to()->item())) 
+	             {    
+				   auto r=ports[1]->units(check);	 
+				   if (!r.empty()) {
+			           vV->setUnits("%"+r.str());
+			           vV->units.normalise();
+                       return vV->units; 
+		           } else {
+					  vV->setUnits("%");
+					  vV->units.normalise();
+                      return vV->units;
+				   } 
```

Comment:
Please try to maintain consistent indenting - this is quite hard to read!

Reply:
Sorry, it appears correctly indented in my editor but does not on github because of tabs. I've removed the tabs and it should appear better now.

Reply:
I don't want to get into the whole tabs vs spaces war, but in my experience, you're better off using spaces everywhere.

Reply:
emacs has an "untabify" command that can help here, if you're still using emacs.